### ISSUE TITLE
No MQTT connection possible

### ISSUE THREAD

**Issue by Razzer90** [Alexandre-io/homeassistant-vcontrol#15]
Since the last update, my MQTT broker no longer gets a connection, even if I enter the data manually. is there a solution for this? ( i use the local MQTT broker. )

**Comment by raebbaer** [comment]
I updated from 1.6.0 to 1.8.1 and the addon is not sending mqtt messages to the broker. With 1.6.0 it worked perfectly. I have the latest HA version. Please repair this bug. Thanks a lot and: Great work!

**Comment by Alexandre-io** [comment]
Hi! I was unable to reproduce this bug. Could you provide me the logs of your addon ? Thanks guys

**Comment by Razzer90** [comment]
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun vclient_pub (no readiness notification)
services-up: info: copying legacy longrun vclient_sub (no readiness notification)
services-up: info: copying legacy longrun vcontrold (no readiness notification)
s6-rc: info: service legacy-services successfully started
[08:37:19] INFO: Setting vcontrold host to 127.0.0.1 and port to 3002 ...
rm: cannot remove '/etc/vcontrold/1_mqtt_commands.txt': No such file or directory
rm: cannot remove '/etc/vcontrold/2_mqtt.tmpl': No such file or directory
[08:37:19] INFO: Setup command 1: getTempA with type FLOAT
[08:37:19] INFO: Setup command 2: getTempWWist with type FLOAT
[08:37:19] INFO: Setup command 3: getTempWWsoll with type FLOAT
[08:37:19] INFO: Setup command 4: getTempKist with type FLOAT
[08:37:19] INFO: Setup command 5: getTempKsoll with type FLOAT
[08:37:19] INFO: Setup command 6: getTempRaumNorSollM1 with type FLOAT
[08:37:19] INFO: Setup command 7: getTempRaumRedSollM1 with type FLOAT
[08:37:19] INFO: Setup command 8: getBrennerStarts with type FLOAT
[08:37:19] INFO: Setup command 9: getBrennerStunden1 with type FLOAT
[08:37:19] INFO: Setup command 10: getPumpeStatusM1 with type FLOAT
[08:37:19] INFO: Setup command 11: getPumpeStatusSp with type FLOAT
[08:37:19] INFO: Setup command 12: getPumpeStatusZirku with type FLOAT
[08:37:19] INFO: Setup command 13: getError0 with type STRING
[08:37:19] INFO: Setup command 14: getError1 with type STRING
[08:37:19] INFO: Setup command 15: getError2 with type STRING
[08:37:19] INFO: Using internal MQTT Host core-mosquitto:1883
[08:37:19] INFO: Setup command 16: getError3 with type STRING
[08:37:19] INFO: Setup command 17: getError4 with type STRING
[08:37:19] INFO: Setup command 18: getError5 with type STRING
[08:37:19] INFO: Using internal MQTT Host core-mosquitto:1883
[08:37:19] INFO: Setup command 19: getError6 with type STRING
[08:37:19] INFO: Setup command 20: getError7 with type STRING
[08:37:19] INFO: Setup command 21: getError8 with type STRING
[08:37:19] INFO: Setup command 22: getError9 with type STRING
[08:37:19] INFO: Setup command 23: getNeigungM1 with type FLOAT
[08:37:19] INFO: Setup command 24: getNiveauM1 with type FLOAT
[08:37:19] INFO: Setup command 25: getBetriebArtM1 with type STRING
[08:37:19] INFO: Setup command 26: getBrennerStufe with type FLOAT
[08:37:19] INFO: Setup command 27: getStatusStoerung with type FLOAT
[08:37:19] INFO: Setup command 28: getBetriebPartyM1 with type FLOAT
[08:37:19] INFO: Setup command 29: getBetriebSparM1 with type FLOAT
[08:37:19] INFO: Setup command 30: getTempPartyM1 with type FLOAT
[08:37:19] INFO: Setting vcontrold host to 127.0.0.1 and port to 3002 ...
[08:37:19] INFO: Setting vcontrold host to 127.0.0.1 and port to 3002 ...



got two mqtt errors. vcontrol works! (in debug mode i can see the states) but mqtt is empty.

**Comment by Schm1tz1** [comment]
@Razzer90 can you provide some debug log (setting DEBUG=true in your config) please? You are connecting to localhost - ist that correct or was some remote setting deleted upon update?

**Comment by Razzer90** [comment]
my settings are the same: 
```
tty: /dev/ttyUSB0
refresh: 30
device_id: "2094"
mqtt_topic: openv
debug: false
commands:
  - getTempA:FLOAT
  - getTempWWist:FLOAT
  - getTempWWsoll:FLOAT
  - getTempKist:FLOAT
  - getTempKsoll:FLOAT
  - getTempRaumNorSollM1:FLOAT
  - getTempRaumRedSollM1:FLOAT
  - getBrennerStarts:FLOAT
  - getBrennerStunden1:FLOAT
  - getPumpeStatusM1:FLOAT
  - getPumpeStatusSp:FLOAT
  - getPumpeStatusZirku:FLOAT
  - getError0:STRING
  - getError1:STRING
  - getError2:STRING
  - getError3:STRING
  - getError4:STRING
  - getError5:STRING
  - getError6:STRING
  - getError7:STRING
  - getError8:STRING
  - getError9:STRING
  - getNeigungM1:FLOAT
  - getNiveauM1:FLOAT
  - getBetriebArtM1:STRING
  - getBrennerStufe:FLOAT
  - getStatusStoerung:FLOAT
  - getBetriebPartyM1:FLOAT
  - getBetriebSparM1:FLOAT
  - getTempPartyM1:FLOAT
vcontrol_host: localhost
vcontrol_port: 3002
```

Debug:
`[71] Mon Jan 30 09:18:08 2023 : TCP socket 0.0.0.0:3002 opened
[71] Mon Jan 30 09:18:08 2023 : Dropped privileges, now running with effective user ID 0, group ID 20
[71] Mon Jan 30 09:18:38 2023 : Client connected 127.0.0.1:54452 (FD:5)
[71] Mon Jan 30 09:18:38 2023 : Command: getTempA
[71] Mon Jan 30 09:18:38 2023 : Process 71 tries to aquire lock
[71] Mon Jan 30 09:18:38 2023 : Process 71 got lock
[71] Mon Jan 30 09:18:38 2023 : >FRAMER: open device /dev/ttyUSB0 ProtocolID 00
[71] Mon Jan 30 09:18:38 2023 : Configuring serial interface /dev/ttyUSB0
[71] Mon Jan 30 09:18:38 2023 : >SENT: 04
[71] Mon Jan 30 09:18:38 2023 : >FRAMER: no preset result
[71] Mon Jan 30 09:18:38 2023 : Waiting for 05
[71] Mon Jan 30 09:18:40 2023 : <RECV: 05 (1670.0 ms)
[71] Mon Jan 30 09:18:40 2023 : >SENT: 01
[71] Mon Jan 30 09:18:40 2023 : >SENT: F7
[71] Mon Jan 30 09:18:40 2023 : >SENT: 08
[71] Mon Jan 30 09:18:40 2023 : >SENT: 00
[71] Mon Jan 30 09:18:40 2023 : >SENT: 02
[71] Mon Jan 30 09:18:40 2023 : >FRAMER: no preset result
[71] Mon Jan 30 09:18:40 2023 : <RECV: len=1 00 (20.0 ms)
[71] Mon Jan 30 09:18:40 2023 : <RECV: len=1 00 (0.0 ms)
[71] Mon Jan 30 09:18:40 2023 : <RECV: received 00 00
[71] Mon Jan 30 09:18:40 2023 : Typ: short (in float: 0.000000)
[71] Mon Jan 30 09:18:40 2023 : (FLOAT) Exp: V/10 [B0:00 B1:00 B2:00 B3:00 B4:00 B5:00 B6:00 B7:00 B8:00 B9:00 BA:00 BB:00 BC:00 BD:00 BE:00 BF:00 ]
[71] Mon Jan 30 09:18:40 2023 : 0.000000 Grad Celsius`
....
`BB:00 BC:00 BD:00 BE:00 BF:00 ]
[71] Mon Jan 30 09:19:40 2023 : 18.000000 
[71] Mon Jan 30 09:19:40 2023 : Command: quit
[71] Mon Jan 30 09:19:40 2023 : Process 71 released lock
[71] Mon Jan 30 09:19:40 2023 : Closed connection (fd:5)
[09:19:40] INFO: Looping vclient. Sleeping 30 seconds.`
works fine!

yes i am on the local host. everything runs in a home assistant instance.

**Comment by Schm1tz1** [comment]
Yes, that was also my idea - good that the +x solved it. Will have a look if we can add handling/error reporting for such cases.

**Comment by Bjoern3003** [comment]
Same Problem here.

Localhost MQTT. And since the Update today (at 15:25) There are still no Updates at mqtt. The Log from vcontrold does not have any errors and the vcontrold is getting Informations with vclient (see it in the Debug Log Info).

I have tried to chmod the 2 files from PR #16 but the result is the same.

**Comment by raebbaer** [comment]
Thanks a lot. Your 1.8.2 release works fine.

**Comment by Bjoern3003** [comment]
Same here, Version 1.8.2 is working fine.

**Comment by Schm1tz1** [comment]
Confirmed - fine with remote setup for both services in 1.8.2

**Comment by Alexandre-io** [comment]
Hi here! Sorry for the glitch introduced into the last release. It should be fixed in 1.8.2.
